openscenario/openscenario_interpreter/include/openscenario_interpreter/syntax/condition.hpp

```diff
@@ -49,9 +57,36 @@ struct Condition : public ComplexType
 
   bool current_value;
 
+private:
+  struct History
+  {
+    double time;
+    bool result;
+  };
+
+  std::list<History> histories;
+
+public:
   explicit Condition(const pugi::xml_node & node, Scope & scope);
 
   auto evaluate() -> Object;
+
+private:
+  template <class R, class... Args>
+  auto update_condition(std::function<R(Args...)> condition) -> Object
+  {
+    histories.push_back({evaluateSimulationTime(), ComplexType::evaluate().as<Boolean>()});
+    if (auto iterator = std::find_if(
+          std::begin(histories), std::end(histories),
+          [this](const auto & entry) { return entry.time > histories.back().time - delay; });
+        std::ptrdiff_t(sizeof...(Args)) <= std::distance(std::begin(histories), iterator)) {
+      current_value = std::apply(condition, std::tuple{Args((--iterator)->result)...});
```

Comment:
C-style casts should not be used.

```suggestion
        static_cast<std::ptrdiff_t>(sizeof...(Booleans)) <= std::distance(std::begin(histories), iterator)) {
      current_value = std::apply(condition, std::tuple(Booleans((--iterator)->result)...));
```

Reply:
Updated to use `static_cast` for `sizeof...(Args)`. But still need to use list-initialization `std::tuple{ ... }` because this line contains side-effect  ( `--iterator` ) and order of evaluation is significant. C++ computes expressions in order from left to right for list-initialization while order of evaluation for arguments in parenthesized initialization is *indeterminately sequenced* (arguments can be evaluated in order from left to right or from right to left).